keps/sig-node/4540-strict-cpu-reservation/README.md

```diff
@@ -0,0 +1,465 @@
+2. The `static` `CPUManager` policy must be selected
+3. The new `strict-cpu-reservation` policy option must be selected
+4. The `reservedSystemCPUs` is not empty
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled by either:
+1. Disabling the `CPUManagerPolicyAlphaOptions` feature gate
+2. Switching the `CPUManager` policy to `none`
+3. Removing `strict-cpu-reservation` from the list of `CPUManager` policy options
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The feature will be enabled regardless it is enabled for the first time or not.
+
+###### Are there any tests for feature enablement/disablement?
+
+- A specific e2e test will demonstrate that the default behaviour is preserved when the feature gate is disabled, or when the feature is not used (2 separate tests)
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Inspect the kubelet configuration -- check the presence of the feature gate and usage of the new policy option.
+
+###### How can someone using this feature know that it is working for their instance?
+
+Inspect the cgroup/cpuset configuration of burstable and best-effort pods -- check the reserved cores are not used by them.
+
+Below is an example when Cgroup v1 is used:
+```console
+# cat /sys/fs/cgroup/cpuset/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod50ca196f\_866c\_4070\_844a\_16d466b957ae.slice/cri-containerd-b511f2ad6e959a0a369578d335c8f0c3f01a2108c9157bce8d239a8d8d6a0d64.scope/cpuset.cpus
+2-19,22-39,42-59,62-79 
```

Comment:
let's try to find something easier to consume and more obvious. This won't be alpha-blocker IMO though

Reply:
agreed, the same checking can be done by kubectl checking the CPU assignment for a container, I will re-write this.